subsystem/core/src/main/java/org/wildfly/extension/camel/service/CamelEndpointDeploymentSchedulerService.java

```diff
@@ -0,0 +1,154 @@
+
+    public CamelEndpointDeploymentSchedulerService() {
+        super();
+    }
+
+    public static ServiceController<CamelEndpointDeploymentSchedulerService> addService(ServiceTarget serviceTarget) {
+        final CamelEndpointDeploymentSchedulerService service = new CamelEndpointDeploymentSchedulerService();
+        return serviceTarget.addService(CamelConstants.CAMEL_ENDPOINT_DEPLOYMENT_SCHEDULER_SERVICE_NAME, service) //
+                .addDependency(GraviaConstants.RUNTIME_SERVICE_NAME, Runtime.class, service.injectedRuntime) //
+                .install();
+    }
+
+    @Override
+    public void start(StartContext context) throws StartException {
+        Runtime runtime = injectedRuntime.getValue();
+        registration = runtime.getModuleContext().registerService(CamelEndpointDeploymentSchedulerService.class, this,
+                null);
+        LOG.warn("*** scheduler started");
+    }
+
+    @Override
+    public void stop(StopContext context) {
+        if (registration != null) {
+            registration.unregister();
+        }
+    }
+
+    @Override
+    public CamelEndpointDeploymentSchedulerService getValue() throws IllegalStateException {
+        return this;
+    }
+
+    /**
+     * Either schedules the given HTTP endpoint for deployment once {@link #deployerService} becomes available or
+     * deploys it instantly if the {@link #deployerService} is available already.
+     *
+     * @param uri
+     *            determines the path and protocol under which the HTTP endpoint should be exposed
+     * @param endpointHttpHandler
+     *            an {@link EndpointHttpHandler} to use for handling HTTP requests sent to the given {@link URI}'s path
+     */
+    public void schedule(URI uri, EndpointHttpHandler endpointHttpHandler) {
+        synchronized (scheduledHandlers) {
+            if (this.deployerService != null) {
+                this.deployerService.deploy(uri, endpointHttpHandler);
+            } else {
+                scheduledHandlers.put(uri, endpointHttpHandler);
+            }
+        }
+    }
+
+    /**
+     * Either removes the given HTTP endpoint from the list of deployments scheduled for deployment or undeploys it
+     * instantly if the {@link #deployerService} is available.
+     *
+     * @param uri
+     *            determines the path and protocol under which the HTTP endpoint should be exposed
+     */
+    public void unschedule(URI uri) {
+        synchronized (scheduledHandlers) {
+            if (this.deployerService != null) {
+                this.deployerService.undeploy(uri);
+            } else {
+                scheduledHandlers.remove(uri);
+            }
+        }
+    }
+
+    /**
+     * Sets the {@link CamelEndpointDeployerService} and deploys any endpoints scheduled for deployment so far.
+     *
+     * @param deploymentService the {@link CamelEndpointDeployerService}
+     */
+    public void setDeploymentService(CamelEndpointDeployerService deploymentService) {
```

Comment:
Perhaps rename this method. It's not really 'setDeploymentService', it's setting the deployment service and possibly other stuff.

Reply:
It is a rich setter :)

`setDeploymentServiceAndDeploy`?

Reply:
Ok lets go with that name. We can always refactor later if needed.

Reply:
done in the next iteration